bench/bench.ml

```diff
@@ -0,0 +1,480 @@
+      nb_entries:int ->
+      log_size:int ->
+      root:string ->
+      name:string ->
+      benchmark ->
+      Benchmark.result =
+   fun ~with_metrics ~nb_entries ~log_size ~root ~name b ->
+    let indices = ref [] in
+    let index_v ~fresh ~readonly n =
+      let i = Index.v ~fresh ~readonly ~log_size (root // name // n) in
+      indices := i :: !indices;
+      i
+    in
+    let result = Benchmark.run ~with_metrics ~nb_entries (b index_v) in
+    !indices |> List.iter (fun i -> Index.close i);
+    result
+
+  type suite_elt = {
+    name : string;
+    synopsis : string;
+    benchmark : benchmark;
+    dependency : string option;
+  }
+
+  let suite =
+    [
+      {
+        name = "replace_random";
+        synopsis = "Replace in random order";
+        benchmark = write_random;
+        dependency = None;
+      };
+      {
+        name = "replace_random_sync";
+        synopsis = "Replace in random order with sync";
+        benchmark = write_sync;
+        dependency = None;
+      };
+      {
+        name = "replace_increasing_keys";
+        synopsis = "Replace in increasing order of keys";
+        benchmark = write_seq;
+        dependency = None;
+      };
+      {
+        name = "replace_increasing_hash";
+        synopsis = "Replace in increasing order of hash";
+        benchmark = write_seq_hash;
+        dependency = None;
+      };
+      {
+        name = "replace_decreasing_hash";
+        synopsis = "Replace in decreasing order of hashes";
+        benchmark = write_rev_seq_hash;
+        dependency = None;
+      };
+      {
+        name = "iter_rw";
+        synopsis = "[RW] Iter";
+        benchmark = iter;
+        dependency = Some "replace_random";
+      };
+      {
+        name = "find_random_ro";
+        synopsis = "[RO] Find in random order";
+        benchmark = find_random_ro;
+        dependency = Some "replace_random";
+      };
+      {
+        name = "find_random_rw";
+        synopsis = "[RW] Find in random order";
+        benchmark = find_random;
+        dependency = Some "replace_random";
+      };
+      {
+        name = "find_absent_ro";
+        synopsis = "[RO] Find absent values";
+        benchmark = find_absent_ro;
+        dependency = Some "replace_random";
+      };
+      {
+        name = "find_absent_rw";
+        synopsis = "[RW] Find absent values";
+        benchmark = find_absent;
+        dependency = Some "replace_random";
+      };
+    ]
+end
+
+let list_benchs () =
+  let pp_bench ppf b = Fmt.pf ppf "%s\t-- %s" b.Index.name b.synopsis in
+  Index.suite |> Fmt.(pr "%a" (list ~sep:Fmt.(const string "\n") pp_bench))
+
+let schedule p s =
+  let todos = List.map fst in
+  let init = ref (s |> List.map (fun b -> (p b.Index.name, b))) in
+  let apply_dep s =
+    let deps =
+      s
+      |> List.fold_left
+           (fun acc (todo, b) ->
+             if todo then
+               match b.Index.dependency with Some s -> s :: acc | None -> acc
```

Comment:
You could do the same thing in `cleanup` to avoid cleaning directories that do not exist.

Reply:
`cleanup` will already check if a file exists before trying to remove it